tests/test_2408_layoutbuilder_in_numba.py

```diff
@@ -0,0 +1,1462 @@
+    builder.clear()
+    assert len(builder) == 0
+
+
+def test_Record_content():
+    @numba.njit
+    def field_index(builder, name):
+        return builder._field_index(name)
+
+    @numba.njit
+    def fill(builder):
+        content_one = builder.content(0)  # "one")
+        content_one.append(1.1)
+        content_two = builder.content(1)  # "two")
+        content_two.append(1)
+        content_three = builder.content(2)  # "three")
+        content_three.append(111)
+
+    builder = lb.Record(
+        [
+            lb.Numpy(np.float64),
+            lb.Numpy(np.int64),
+            lb.Numpy(np.uint8, parameters={"__array__": "char"}),
+        ],
+        ["one", "two", "three"],
+    )
+    assert field_index(builder, "one") == 0
+    assert field_index(builder, "two") == 1
+    assert field_index(builder, "three") == 2
+
+    with pytest.raises(ValueError):
+        field_index(builder, "four")  # ValueError: tuple.index(x): x not in tuple
+
+    fill(builder)
+    array = builder.snapshot()
+    assert ak.to_list(array) == [{"one": 1.1, "three": "o", "two": 1}]  # ???
+
+    error = ""
+    assert builder.is_valid(error), error
+
+    assert len(builder) == 1
+    builder.clear()
+    assert len(builder) == 0
+
+
+def test_Tuple_append():
+    @numba.njit
+    def f38(builder):
+        content_one = builder.content(0)
+        content_one.append(1.1)
+        content_two = builder.content(1)
+        content_list = content_two.begin_list()
+        content_list.append(1)
+        content_list.append(2)
+        content_list.append(3)
+        content_two.end_list()
+
+    builder = lb.Tuple(
+        [lb.Numpy(np.float64), lb.ListOffset(np.int64, lb.Numpy(np.int32))]
+    )
+    f38(builder)
+    array = builder.snapshot()
+    assert ak.to_list(array) == [(1.1, [1, 2, 3])]
+
+    error = ""
+    assert builder.is_valid(error), error
+
+    assert len(builder) == 1
+    builder.clear()
+    assert len(builder) == 0
+
+
+def test_Union_append():
+    @numba.njit
+    def f39(builder):
+        one = builder.append_content(0)
+        one.append(1.1)
+
+        two = builder.append_content(1)
+        list = two.begin_list()
+        list.append(1)
+        list.append(2)
+        list.append(3)
+        two.end_list()
+
+    builder = lb.Union(
+        np.int8,
+        np.int64,
+        [
+            lb.Numpy(np.float64),
+            lb.ListOffset(np.int64, lb.Numpy(np.int32)),
+        ],
+    )
+    assert len(builder) == 0
+
+    f39(builder)
+
+    array = builder.snapshot()
+    assert ak.to_list(array) == [1.1, [1, 2, 3]]
+
+
+def test_numba_append():
+    @numba.njit
+    def create():
+        return lb.Numpy(np.int32)
+
+    @numba.njit
+    def append_range(builder, start, stop):
+        for x in range(start, stop):
+            builder.append(x)
+
+    @numba.njit
+    def append_single(builder, x):
+        builder.append(x)
+
+    @numba.njit
+    def snapshot(builder):
+        return builder.snapshot()
+
```

Comment:
Perform the `snapshot` outside of the compiled context.

```suggestion
```

Reply:
done